platform/macos/macos_terminal_logger.mm

```diff
@@ -46,37 +46,46 @@
 		err_details = p_code;
 	}
 
+	const char *indent = "";
 	switch (p_type) {
 		case ERR_WARNING:
+			indent = "     ";
 			os_log_info(OS_LOG_DEFAULT,
 					"WARNING: %{public}s\nat: %{public}s (%{public}s:%i)",
 					err_details, p_function, p_file, p_line);
 			logf_error("\E[1;33mWARNING:\E[0;93m %s\n", err_details);
-			logf_error("\E[0;90m     at: %s (%s:%i)\E[0m\n", p_function, p_file, p_line);
+			logf_error("\E[0;90m%sat: %s (%s:%i)\E[0m\n", indent, p_function, p_file, p_line);
 			break;
 		case ERR_SCRIPT:
+			indent = "          ";
 			os_log_error(OS_LOG_DEFAULT,
 					"SCRIPT ERROR: %{public}s\nat: %{public}s (%{public}s:%i)",
 					err_details, p_function, p_file, p_line);
 			logf_error("\E[1;35mSCRIPT ERROR:\E[0;95m %s\n", err_details);
-			logf_error("\E[0;90m          at: %s (%s:%i)\E[0m\n", p_function, p_file, p_line);
+			logf_error("\E[0;90m%sat: %s (%s:%i)\E[0m\n", indent, p_function, p_file, p_line);
 			break;
 		case ERR_SHADER:
+			indent = "          ";
 			os_log_error(OS_LOG_DEFAULT,
 					"SHADER ERROR: %{public}s\nat: %{public}s (%{public}s:%i)",
 					err_details, p_function, p_file, p_line);
 			logf_error("\E[1;36mSHADER ERROR:\E[0;96m %s\n", err_details);
-			logf_error("\E[0;90m          at: %s (%s:%i)\E[0m\n", p_function, p_file, p_line);
+			logf_error("\E[0;90m%sat: %s (%s:%i)\E[0m\n", indent, p_function, p_file, p_line);
 			break;
 		case ERR_ERROR:
 		default:
+			indent = "   ";
 			os_log_error(OS_LOG_DEFAULT,
 					"ERROR: %{public}s\nat: %{public}s (%{public}s:%i)",
 					err_details, p_function, p_file, p_line);
 			logf_error("\E[1;31mERROR:\E[0;91m %s\n", err_details);
-			logf_error("\E[0;90m   at: %s (%s:%i)\E[0m\n", p_function, p_file, p_line);
+			logf_error("\E[0;90m%sat: %s (%s:%i)\E[0m\n", indent, p_function, p_file, p_line);
 			break;
 	}
+
+	for (const Ref<ScriptBacktrace> &backtrace : p_script_backtraces) {
+		logf_error("\E[0;94m%s\E[0m\n", backtrace->format(strlen(indent)).utf8().get_data());
```

Comment:
```suggestion
		os_log_error(OS_LOG_DEFAULT, "%{public}s", backtrace->format().utf8().get_data());
		logf_error("\E[0;94m%s\E[0m\n", backtrace->format(strlen(indent)).utf8().get_data());
```

Reply:
This would print the backtrace twice. The headers for the backtraces is part of the `format` call, so there's no way to do `os_log_error` separately for that part, if that was what you were hoping to do.

I left this as-is.

Reply:
`os_log_error` sends it to message log in `Console.app`, it's not printing anything. So fully duplicating backtrack is exactly what's this supposed to do.

`logf_error` ⇾ prints.
`os_log_error` ⇾ sends to OS log.

But ideally it should be appended directly to the argument of the previous `os_log_error` (in the switch), to send the whole thing as a one message.

Reply:
Not sure how I missed that all the other log messages were being duplicated like this as well.

I've made the suggested change. Passing it as part of the preceding `os_log_error` got a bit messy with having to iterate over the `Vector` and stuff, so I opted not to.